make/platformio/platformio.py

```diff
@@ -407,20 +408,36 @@ def setup_board_nano32(env):
 for src in env.LookupSources(variant_dir, src_dir, True, src_filter):
     source_files.append(env.Object(src))
 
-# Command to generate simba_gen.c
-env.Command(SIMBA_GEN_C,
-            source_files,
-            ('"$PYTHONEXE" "$PLATFORMFW_DIR/bin/simbagen.py" '
+# action string to generate simba_gen.c
+gen_command_action = ('"$PYTHONEXE" "$PLATFORMFW_DIR/bin/simbagen.py" '
              '--output-directory "$BUILD_DIR/SimbaFramework" '
              'source '
              '--name "$NAME" '
              '--version "$VERSION" '
              '--board "$BOARD_DESC" '
              '--mcu "$MCU_DESC" '
              '--endianess little '
-             '--eeprom-soft-chunk-size 2048 '))
+             '--eeprom-soft-chunk-size 2048 ')
+
+# get settings file path
+settings_ini = ARGUMENTS.get('SETTINGS_INI')
+if settings_ini is not None:
+    try:
+        settings_ini = str(env.File(settings_ini.decode('base64'), 
+                                    env['PROJECT_DIR']))
+    except Exception, e:
```

Comment:
Hmm, ``Exception, e`` is the old Python syntax, the new is ``Exception as e``. I guess it's good to use the new syntax.